homa_incoming.c

```diff
@@ -683,20 +683,20 @@ void homa_resend_pkt(struct sk_buff *skb, struct homa_rpc *rpc,
 
 	if (!homa_is_client(rpc->id) && rpc->state != RPC_OUTGOING) {
 		/* We are the server for this RPC. */
-		if ((rpc->msgin.length - rpc->msgin.bytes_remaining) >= rpc->msgin.granted) {
-			/* We've received everything that we've granted, so either the server 
-			   is busy with an influx of messages exceeding max_incoming, or a 
-			   grant packet has been dropped. In either case, the server should 
-			   initiate RESEND
-			   */
+		if ((rpc->msgin.length - rpc->msgin.bytes_remaining) 
+				>= rpc->msgin.granted) {
+			/* We've received everything that we've granted, so likely we
+			 * are busy with an influx of messages exceeding max_incoming.
+			 * We should send a BUSY right now and initiate a RESEND
+			 * later when we are free.
+			 */
```

Comment:
This comment isn't quite right. For example, there isn't necessarily "an influx of messages exceeding max_incoming". Also, I don't think "later when we are free" is correct; the conditions for sending a RESEND later are somewhat complicated.

It feels like you're overdocumenting: don't try to describe things that are happening (or will happen) elsewhere in the system unless they are essential to understanding what's happening here. Such documentation tends to become stale as the system evolves. In this particular case, I'd suggest changing the sense of the "if" to switch the "then" and "else" clauses, then replace the comment on line 685 with this: "We are the server for the RPC. If we haven't received all of the bytes we've granted then request a resend of the missing bytes; otherwise just send a BUSY."

But, see my comment below for a more radical change.